### PR TITLE
Acquisition Rate of planar mode is not used

### PR DESCRIPTION
**Pull request by fedem-p**
Here's a simple fix to the second problem listed in #144.

## The Issue
changing the parameter regarding the how many planes per second should be recorded doesn't update the saving parameter values

The top bar gui receive the info to display (number of volumes) from the save_status inside state.py
The save_status is changed in the stacksaver using the received save_params
the save_params is a property of the state which converts the appropriate settings in saving parameters

```mermaid
flowchart LR;
    stacksaver -->save_status;
    save_params -->stacksaver;
    save_status -->top_bar_gui;

    subgraph state
settings --> save_params;
    save_status;
    save_params;
    end 

    subgraph top_guy
    top_bar_gui;
    end

    subgraph stream_saver
    stacksaver;
    end
```

Now, the problem relies in the fact that `save_params` and the function `convert_save_params` only accepted/used `ZRecordingSettings` to extract the scanning_settings.
This works well for volumetric mode but fails to account for any changes in the single_plane_settings.

## The Solution
I changed three functions:
1. `save_params` to determine whether it's planar or volumetric mode and pass to `convert_save_params` the appropriate settings (the default is volumetric)
2. `convert_save_params` to accept and correctly extract information from both volume_settings and single_plane_settings (in particular -> n_planes is set to 0 if in planar mode)
3. `get_voxel_size` to compute a voxel size even in planar_mode -> z voxel is left at 1 

## What's left
I'd like to hear some feedback regarding two main values:
- `n_planes` which is set to 0 in planar_mode -> maybe it could be better to set it to None?
- the default `voxel size` for the z axis in planar mode is set to 1 -> or maybe it should only return a tuple of 2 values if it's planar mode?
- unfortunately this doesn't solve the first problem listed in #144 which would now affect also the planar mode (probably better addressed in another PR)
- [ ] write docstrings

### CODE REVIEW

File: sashimi/state.py [hunk 1]
```diff
@@ -224,16 +225,20 @@ def calculate_calibration(self):
...
 
-    inter_plane = scan_length / scanning_settings.n_planes
+    if isinstance(scanning_settings, SinglePlaneSettings):
+        inter_plane = 1
+    else:
+        scan_length = (
+            scanning_settings.piezo_scan_range[1]
+            - scanning_settings.piezo_scan_range[0]
+        )
+        inter_plane = scan_length / scanning_settings.n_planes
```
Comment (fedem-p) [inline sashimi/state.py:241]: Inside `get_voxel_size`:
checks if the scanning_settings are single_plane_settings and if so sets the z_voxel size to 1

File: sashimi/state.py [hunk 2]
```diff
@@ -244,13 +249,16 @@ def get_voxel_size(
...
 ):
-    n_planes = scanning_settings.n_planes - (
-        scanning_settings.n_skip_start + scanning_settings.n_skip_end
-    )
+    if isinstance(scanning_settings, SinglePlaneSettings):
+        n_planes = 0
+    else:
+        n_planes = scanning_settings.n_planes - (
+            scanning_settings.n_skip_start + scanning_settings.n_skip_end
+        )
```
Comment (fedem-p) [inline sashimi/state.py:261]: Inside `convert_save_params` checks if the settings are for planar mode, and if so,  sets n_planes to 0

File: sashimi/state.py [hunk 3]
```diff
@@ -472,12 +480,21 @@ def n_planes(self):
...
+                self.trigger_settings,
+            )
+        else:
+            save_p = convert_save_params(
+                self.save_settings,
+                self.volume_setting,
+                self.camera_settings,
+                self.trigger_settings,
+            )
+        return save_p
```
Comment (fedem-p) [inline sashimi/state.py:497]: inside `save_params` checks if we are in planar mode and returns the planar settings - if not defaults to volume settings

**Comment by diegoasua** [comment]
Let's keep `n_planes` as 0 so that there is no casting. And it is actually more informative. On the other hand `Int` to `None` is not exactly a casting operation but it could have some undesirable effects downstream. In particular, I am not 100% sure that the [`external_communication`](https://github.com/portugueslab/sashimi/blob/aaffce7eb6228da42ce929d0dc106f76000f90e2/sashimi/utilities.py#L43) process will handle this correctly. It should. And testing should catch that otherwise. But certainty is not an exact science in Sashimi as you know. And overall it makes more sense to specify that there were 0 planes than `None` planes, as long as this is documented as being a single-plane recording. Same goes for voxel size, just setting to 1 is fine.

TLDR keep `n_planes` as 0

**Review by diegoasua — APPROVED** [review]

**Comment by diegoasua** [comment]
By the way that diagram is super cool. Which tool did you use?

**Comment by vigji** [comment]
I would say that the defaults you use would depend on the saving format.
 - If the output in planar mode is a `(t, 1, w, h)` stack, then `n_planes` should be 1 and `voxel_size=(None, w_res, h_res)`
 - If the output in planar mode is a `(t, w, h)` stack, then `n_planes` should be `None`/0 and `voxel_size=(w_res, h_res)`

**Comment by fedem-p** [comment]
> By the way that diagram is super cool. Which tool did you use?

I used a github codeblock with mermaid (I literally found it yesterday and is basically a languagge for writing flowcharts and other cool things) :)

**Comment by nvbln** [comment]
Since there have been no comments in the last two years and the PR seems to work I'll merge it.
